src/cdomains/lval.ml

```diff
@@ -382,25 +388,38 @@ struct
 end
 
 (** Lvalue lattice with sublattice representatives for {!DisjointDomain}. *)
-module NormalLatRepr (Idx: IntDomain.Z) =
+module NormalLatRepr (Idx: IdxDomain) =
 struct
   include NormalLat (Idx)
 
+  module UnitIdxDomain =
+  struct
+    include Lattice.Unit
+    let equal_to _ _ = `Top
+    let to_int _ = None
+  end
   (** Representatives for lvalue sublattices as defined by {!NormalLat}. *)
   module R: DisjointDomain.Representative with type elt = t =
   struct
-    include Normal (Idx)
     type elt = t
 
-    let rec of_elt_offset: Offs.t -> Offs.t =
+    (* Offset module for representative without abstract values for index offsets, i.e. with unit index offsets.
+       Reason: The offset in the representative (used for buckets) should not depend on the integer domains,
+       since different integer domains may be active at different program points. *)
+    include Normal (UnitIdxDomain)
+
+    let rec of_elt_offset (*: (fieldinfo, Idx.t) offs -> (fieldinfo, UnitIdxDomain.t) offs *) =
```

Comment:
Does this type signature not work anymore or?

Reply:
It actually works, so I re-added it.